### PR TITLE
Add vertical navigation, split clusters/storages/plans into separate pages, remove dashboard cards

### PR DESCRIPTION
**Pull request by mturley**
This PR adds routes and navigation, removes the summary cards, and splits the current home page into 3 new pages by resource (Clusters, Replication repositories, and Migration plans). The existing DetailViewComponent is broken into separate redux-connected components for each page, splitting up the selectors based on what is necessary for each resource. Each DataList is now replaced by a table with filters and pagination. Column structure and icons are based on @vconzola 's mockups [here](https://docs.google.com/presentation/d/1JU_Lw1A-gUO4gBRofZzi74aONcknYRHbqBl_xHxikYQ/edit#slide=id.g8482032e20_0_3) and Mary Shakshober's mockups [here](https://marvelapp.com/58cgihg/screen/61031650).

Since the PlanContext, ClusterContext and StorageContext were already separate and we already had react-router set up, most of the work involved converting the DataLists to Tables, moving/refactoring the descendant components into page-specific places, and adding improved TypeScript types.

I considered also splitting up the polling currently managed in AppComponent, so that we aren't polling all three types of resources on all 3 pages. However (a) I think we may just want to leave that as-is, so everything is pre-loaded and users can navigate between pages quickly and (b) even if we decide to split up the polling, it will probably be fairly easy because it's already organized by resource.

# Screens

## Clusters page, at `/clusters`:

Screenshot 2020-06-01 21 18 37

## Clusters page with the left navigation closed:

Screenshot 2020-06-01 21 20 18

## Replication repositories page, at `/storages`:

Screenshot 2020-06-01 21 25 59

## Migration plans page, at `/plans`:

Screenshot 2020-06-01 21 19 10

### CODE REVIEW

**Comment by mturley** [comment]
Kudos to @ibolton336 for keeping a nice separation of concerns along resource type lines, it made this a lot easier than I expected!

**Comment by mturley** [comment]
@ibolton336 @eriknelson are you guys ok with the approach I'm taking here (leaving all the polling combined in AppComponent, driving the `/` route and all its children from HomeComponent, defining sub-routes in there inside the page layout)? Mostly I want to determine if we need to separate the polling by which page you're on, since that will affect how I structure those components a bit.

**Comment by ibolton336** [comment]
@mturley I think it looks great. I am on board with the direction you are taking it. I agree it would make more sense to break up the DetailViewComponent into connected page components. I dont see an issue with leaving the polling controls at the global app level component for now either since you can still access the start/stop methods from the polling context at any level.

**Comment by mturley** [comment]
Awesome! I'm almost done breaking up the separate connected page components, and I'm meeting with Vince tomorrow to talk about the visual changes.

**Comment by mturley** [comment]
Clusters page is fully converted to a table, the other pages are in progress. After that all I need to do is fix the TypeScript annotations and this should be good to go.
Screenshot 2020-05-26 17 27 35

File: src/app/home/pages/PlansPage/components/PlansTable.tsx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+            {
+              title: <PlanActions plan={plan} />,
+              props: {
+                className: 'pf-c-table__action',
+              },
+            },
+          ],
+        },
+        {
+          parent: planIndex * 2,
```
Comment (ibolton336) [inline src/app/home/pages/PlansPage/components/PlansTable.tsx:121]: This line still makes my head hurt :)

File: src/app/home/pages/ClustersPage/ClustersPage.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+              {!clusterList ? null : clusterList.length === 0 ? (
+                <EmptyState variant="full">
+                  <EmptyStateIcon icon={AddCircleOIcon} />
+                  <Title size="lg">Add source and target clusters for the migration</Title>
+                  <Button onClick={toggleAddEditModal} variant="primary">
+                    Add cluster
+                  </Button>
+                </EmptyState>
+              ) : (
+                <ClustersTable
```
Comment (ibolton336) [inline src/app/home/pages/ClustersPage/ClustersPage.tsx:63]: +1 Really awesome to see us move to tables here. Makes navigating the tree much easier.

File: src/app/home/pages/ClustersPage/components/ClustersTable.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    const { clusterName, clusterUrl, associatedPlanCount, clusterStatus } = getClusterInfo(
+      cluster,
+      migMeta,
+      associatedPlans
+    );
+    return [clusterName, clusterUrl, associatedPlanCount, clusterStatus, ''];
+  };
+
+  const { sortBy, onSort, sortedItems } = useSortState(clusterList, getSortValues);
+  const { currentPageItems, setPageNumber, paginationProps } = usePaginationState(sortedItems, 10);
```
Comment (ibolton336) [inline src/app/home/pages/ClustersPage/components/ClustersTable.tsx:40]: Really cool use of hooks here. The composition pattern is very clean & it is great to be able to pull from common hooks across the entire app. Great Job!

Also, +1 for consistency between the pages & your pattern of extracting derived state/data formatting to helper functions.

File: src/app/home/pages/PlansPage/components/PlansTable.tsx (hunk 1 above)
Comment (mturley) [reply src/app/home/pages/PlansPage/components/PlansTable.tsx:121]: ha, yeah, I think maybe it's worth at least adding a comment there. I'll think about how we can make that more readable.

**Comment by mturley** [comment]
Thanks for the review @ibolton336! I plan to push TypeScript annotations asap to wrap this up, I'll ping you for a re-review when that's ready. I don't expect any of the structure to change.

File: src/app/home/pages/PlansPage/components/PlansTable.tsx (hunk 1 above)
Comment (mturley) [reply src/app/home/pages/PlansPage/components/PlansTable.tsx]: Added a comment to clarify the math.
```
// Plan 0 => rows 0 and 1, Plan 1 => rows 2 and 3, Plan 2 => rows 4 and 5, etc.
```

**Comment by mturley** [comment]
@ibolton336 this is ready for a final review now! It took me another 11 commits to add all the types, sorry... I also rebased the branch. Here's a diff of only the changes since your review: https://github.com/konveyor/mig-ui/compare/3fff058...mturley:add-vertical-nav

**Comment by mturley** [comment]
Marking `do-not-merge` per @eriknelson's instructions this morning that we wait for the 1.2.1 branch to be cut before merging this to master.

**Comment by eriknelson** [comment]
@mturley awesome, I was talking to Ian earlier today, I think we're ready to cut a release branch tomorrow and bring this in asap afterwards.

**Pull request merged**